src/js/containers/search/filters/recipient/RecipientLocationContainer.jsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * RecipientLocationContainer.jsx
+ * Created by michaelbray on 2/17/17.
+ */
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as recipientActions from 'redux/actions/search/recipientActions';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+const propTypes = {
+    toggleRecipientLocation: React.PropTypes.func,
+    setAutocompleteRecipientLocations: React.PropTypes.func,
+    selectedRecipientLocations: React.PropTypes.object,
+    locationDomesticForeign: React.PropTypes.string,
+    autocompleteRecipientLocations: React.PropTypes.array
+};
+
+export class RecipientLocationContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            recipientLocationSearchString: '',
+            autocompleteRecipientLocations: []
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.timeout = null;
+    }
+
+    componentDidMount() {
+        this.parseAutocompleteRecipientLocations(this.props.autocompleteRecipientLocations);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(nextProps.autocompleteRecipientLocations,
+                this.props.autocompleteRecipientLocations)) {
+            this.parseAutocompleteRecipientLocations(nextProps.autocompleteRecipientLocations);
+        }
+    }
+
+    parseAutocompleteRecipientLocations(recipientLocations) {
+        const values = [];
+        if (recipientLocations.length > 0) {
+            recipientLocations.forEach((item) => {
+                let placeType = _.upperCase(item.place_type);
+                if (item.parent !== null) {
+                    placeType += ` in ${item.parent}`;
+                }
+
+                values.push({
+                    title: item.place,
+                    subtitle: placeType,
+                    data: item
+                });
+            });
+        }
+
+        this.setState({
+            autocompleteRecipientLocations: values
+        });
+    }
+
+    queryAutocompleteRecipientLocations(input) {
+        // Only search if input is 2 or more characters
+        if (input.length >= 2) {
+            this.setState({
+                recipientLocationSearchString: input
+            });
+
+            if (this.recipientLocationSearchRequest) {
+                // A request is currently in-flight, cancel it
+                this.recipientLocationSearchRequest.cancel();
+            }
+
+            const locSearchParams = {
+                value: this.state.recipientLocationSearchString,
+                scope: this.props.locationDomesticForeign,
```

Comment:
Scope is not working, it appears this prop should be named `recipientDomesticForeign`.